src/ast/nodes/UnaryExpression.ts

```diff
@@ -69,4 +83,56 @@ export default class UnaryExpression extends NodeBase {
 			this.scope.context.requestTreeshakingPass();
 		}
 	}
+
+	include(
+		context: InclusionContext,
+		includeChildrenRecursively: IncludeChildren,
+		_options?: InclusionOptions
+	): void {
+		if (!this.deoptimized) this.applyDeoptimizations();
+		this.included = true;
+		const literalValue = includeChildrenRecursively
+			? UnknownValue
+			: this.getLiteralValueAtPath(EMPTY_PATH, SHARED_RECURSION_TRACKER, this);
+		if (typeof literalValue === 'symbol' || this.argument.shouldBeIncluded(context)) {
+			this.argument.include(context, includeChildrenRecursively);
+			this.literalValue = UnknownValue;
+		} else {
+			this.literalValue = literalValue;
+		}
+	}
+
+	render(code: MagicString, options: RenderOptions) {
+		if (typeof this.literalValue === 'symbol') {
+			super.render(code, options);
+		} else {
+			code.overwrite(this.start, this.end, getRenderString(this.literalValue));
+		}
+	}
+}
+
+function getRenderString(value: unknown) {
+	if (value === undefined) {
+		return 'void 0';
+	}
+	if (typeof value === 'string') {
+		return JSON.stringify(value);
+	}
+	if (typeof value === 'number') {
+		return simplifyNumber(value);
+	}
+	return String(value);
+}
+
+function simplifyNumber(value: number) {
+	if (Object.is(-0, value)) {
+		return '-0';
+	}
+	if ((Math.abs(value) > 1000 && value % 1000 === 0) || (value !== 0 && Math.abs(value) < 0.01)) {
+		const exp = value.toExponential();
+		const [base, exponent] = exp.split('e');
+		const floatLength = base.split('.')[1]?.length || 0;
+		return `${base.replace('.', '')}e${parseInt(exponent) - floatLength}`;
+	}
+	return String(value).replace('+', '');
```

Comment:
Is there really a case where `String(value)` still has a `+`?

Reply:
if the value is `1e28`, the result will be `1e+28`